diff -u --recursive hpklinux_4.20.39-orig/hpi-drv/hpi6205.c hpklinux_4.20.39/hpi-drv/hpi6205.c
--- hpklinux_4.20.39-orig/hpi-drv/hpi6205.c	2023-02-15 12:07:46.000000000 -0500
+++ hpklinux_4.20.39/hpi-drv/hpi6205.c	2023-08-10 17:00:04.547300253 -0400
@@ -707,9 +707,9 @@
 	struct hpi_hw_obj_62 *phw = pao->priv;
 	int i;
 
-	i = pci_set_dma_mask(pci_dev, DMA_BIT_MASK(32));
+	i = dma_set_mask(&pci_dev->dev, DMA_BIT_MASK(32));
 	if (!i)
-		i = pci_set_consistent_dma_mask(pci_dev, DMA_BIT_MASK(32));
+		i = dma_set_coherent_mask(&pci_dev->dev, DMA_BIT_MASK(32));
 
 	if (i) {
 		dev_err(&pci_dev->dev, "DMA set 32 bit mask error %d\n", i);
diff -u --recursive hpklinux_4.20.39-orig/hpi-drv/hpi6700.c hpklinux_4.20.39/hpi-drv/hpi6700.c
--- hpklinux_4.20.39-orig/hpi-drv/hpi6700.c	2023-02-15 12:07:47.000000000 -0500
+++ hpklinux_4.20.39/hpi-drv/hpi6700.c	2023-08-10 18:09:48.195287465 -0400
@@ -1385,9 +1385,9 @@
 	struct hpi_hw_obj_67 *phw = pao->priv;
 	int i;
 
-	i = pci_set_dma_mask(pci_dev, DMA_BIT_MASK(32));
+	i = dma_set_mask(&pci_dev->dev, DMA_BIT_MASK(32));
 	if (!i)
-		i = pci_set_consistent_dma_mask(pci_dev, DMA_BIT_MASK(32));
+		i = dma_set_coherent_mask(&pci_dev->dev, DMA_BIT_MASK(32));
 	if (i) {
 		dev_err(&pci_dev->dev, "DMA set 32 bit mask error %d\n", i);
 		return HPI_ERROR_MEMORY_ALLOC;
